#103 - Implement convex hull for an array of sets

mforets opened the pull request:
Closes #103.

schillic commented on src/ConvexHull.jl:
```diff
@@ -106,3 +107,52 @@ function σ(d::AbstractVector{<:Real}, ch::ConvexHull)::AbstractVector{<:Real}
+    array::Vector{S}
+end
+# type-less convenience constructor
+ConvexHullArray(a::Vector{S}) where {S<:LazySet{N}} where {N<:Real} = ConvexHullArray{N, S}(a)
```
Is this needed when there is no inner constructor? I remember that sometimes you need it...

mforets replied:
yes, its' needed, just checked

schillic commented on src/ConvexHull.jl:
```diff
@@ -106,3 +107,52 @@ function σ(d::AbstractVector{<:Real}, ch::ConvexHull)::AbstractVector{<:Real}
+       s[i] = σ(d, ci)
+       r[i] = dot(d, s[i])
+    end
+    return s[indmax(r)]
```
Are you sure about this? What about negative directions?

mforets commented:
i changed the algorithm for σ(d, cha) (only keep greatest value of support vector), and added some examples. the idea of the second one was the plot (but i don't want to add `using Plots`)

```julia
b = [Ball2([2*pi*i/100, sin(2*pi*i/100)], 0.01) for i in 1:100];
c = ConvexHullArray(b);
plot(c, 1e-3, alpha=0.2, color="red")
plot!(b, 1e-3, alpha=0.2, color="orange")
```

maybe i'll propose to add it to the [Convex hulls](https://juliareach.github.io/LazySets.jl/latest/man/convex_hulls.html) manual section.

mforets commented:
done, this is ready to go..

schillic commented:
It is a bit inconvenient that we do not have a function symbol for the convex hull. Now if you want to add a new set to a `ConvexHullArray`, you have to access the underlying array. Should we say that `CH(cha::ConvexHullArray, X::LazySet) = cha.array.add(X)` (+ the symmetric case + the *merge* case for two `ConvexHullArray`s)?

schillic commented on src/ConvexHull.jl:
```diff
@@ -164,6 +164,18 @@ Alias for `ConvexHullArray`.
 const CHArray = ConvexHullArray
 
+CH(cha::ConvexHullArray, ∅::EmptySet) = cha
+CH(cha::EmptySet, ::ConvexHullArray) = cha
```
`cha` should be the *second* argument.

schillic commented on src/ConvexHull.jl:
```diff
@@ -164,6 +164,18 @@ Alias for `ConvexHullArray`.
+CH(cha::EmptySet, ::ConvexHullArray) = cha
+
+function CH(cha::ConvexHullArray, X::LazySet)
+    push!(cha.array.X)
```
How does this work? Is it not `push!(cha.array, X)`?

mforets commented:
the `CH(cha::ConvexHullArray, X::LazySet)` requires i think to change the def of `CH` so i'm leaving this outside this issue, see #165.
